WordPress/gutenberg
# Dismiss certain pasted files depending on the HTML payload

This pull request seeks to fix regressions introduced by #38459, which itself sought to patch #4882.

## Description

When content is pasted into RichText components, Gutenberg needs to decide what to paste when multiple sources are present in the same clipboard: plain text, HTML source, and attached files.

#4882 originally worked around a quirk of MS Word by ignoring attached files if there was an HTML payload in the clipboard, because Word was actually attaching "screenshots" of the copied content. The workaround was enough for a few years, until other applications started to more heavily rely on multi-format pastes (e.g. in Google Photos, copying an image also results in some fallback HTML content being copied). #38459 recently fixed that by specifically detecting MS Office content instead of the mere presence of HTML content.

However, there are now reports suggesting new false negatives (i.e. users who paste content from MS Office are expecting to see new paragraphs but are seeing a new image). I then found out that other office suites are now using the same trick of attaching rendered screenshots upon copying content (e.g. Apple Numbers), _so we need to switch from a strategy of targeting quirky applications to making decisions based on the shape of the clipboard_.

This PR proposes a simple heuristic to determine whether specific pasted images are to be dismissed. In order for the heuristic to be good, we need to collect concrete use cases in the form of unit tests ([source](https://github.com/WordPress/gutenberg/pull/38992/files#diff-c2a857f4a5abd48b33db3ed6ee0ab23872bfdd6a9fa45d84b66e939c17a37c0f)). **Any new use cases are welcome.**

## Testing Instructions

1. Follow instructions at https://github.com/WordPress/gutenberg/pull/38992#issuecomment-1048103568 for specific edge cases.
2. Follow instructions at https://github.com/WordPress/gutenberg/pull/38459 to ensure no regressions

## Screenshots 

## Types of changes





## Checklist:
